guide/docs/popular-topics/permissions.mdx

```diff
@@ -1,8 +1,384 @@
+    moderate_members=True,
+)
+
+chatmod_role = await guild.create_role(
+    name="Chat Moderator",
+    permissions=permissions,
+    hoist=True,
+)
+```
+
+Creating an `#only-mods` <DocsLink reference="disnake.TextChannel"/>:
+
+```python
+# Allow mods to see the channel
+overwrite = disnake.PermissionOverwrite(view_channel=True)
+# See `TextChannel.edit()` documentation.
+mapping = {chatmod_role: overwrite}
+
+only_mods_channel = await guild.create_text_channel("only-mods", overwrites=mapping)
+
+assert not only_mods_channel.permissions_for(guild.default_role).view_channel
+assert only_mods_channel.permissions_for(chatmod_role).view_channel
+```
+
+A slash command that checks whether the command author can add reactions in the current channel:
+
+```python
+@commands.slash_command(description="Checks whether you can add reactions!")
+async def can_add_reactions(inter: disnake.ApplicationCommandInteraction) -> None:
+    await inter.send("Sure you can!" if inter.permissions.add_reactions else "Sadly no :(")
+```
+
+## Querying permissions
+
+There are various ways to query permission information, and they also depend on the context you're
+in. This section tries to list all possible methods for getting permission info, grouped by context
+they can be used in.
+
+### Context-agnostic
+
+If you have an instance of any class implementing <DocsLink reference="disnake.abc.GuildChannel" />,
+then the easiest way to query permissions for a member/role is
+through <DocsLink reference="disnake.abc.GuildChannel.permissions_for" />.
+This also works with <DocsLink reference="disnake.Thread" />s.
+If you need a bit more lower level control, you can also get raw overwrite objects
+using <DocsLink reference="disnake.abc.GuildChannel.overwrites_for" />
+or <DocsLink reference="disnake.abc.GuildChannel.overwrites" />.
+
+### In prefix commands
+
+There are no direct methods on <DocsLink reference="disnake.ext.commands.Context" /> for querying permission
+information, however, you can still access <DocsLink reference="disnake.ext.commands.Context.channel" /> and
+call above-mentioned methods on it, though be sure to check that it's not <DocsLink reference="disnake.PartialMessageable" />
+as it doesn't have any permissions-related methods on it either.
+
+### In interactions
+
+When working with interactions (i.e., with application commands or components), there are two main properties
+defined on <DocsLink reference="disnake.Interaction" />: <DocsLink reference="disnake.Interaction.permissions" />
+(for getting the interaction author's permissions in the current channel) and <DocsLink reference="disnake.Interaction.app_permissions" />
+(for getting the bot's permissions).
+
+## How permissions are calculated
+
+Permissions can be a tough thing to understand at first, but understanding how permissions are
+applied can help you <!-- feel the pain of us mere mortals re-implementing discord client's obscure behaviour -->
+better grasp this concept.
```

Comment:
```suggestion
Permissions can be a tough thing to understand at first, but understanding how permissions are
applied can help you better grasp this concept.
```
github's syntax highlighter really doesn't like this for some reason, so we should probably just remove the comment c:
it do be like that though.